src/main/groovy/com/twcable/grabbit/GrabbitConfiguration.groovy

```diff
@@ -117,6 +119,31 @@ class GrabbitConfiguration {
         )
     }
 
+    /**
+     * Removes '/.' from {@code pathConfigurations}'s path if exists in the end, and Sorts the {@code pathConfigurations}
+     * lexicographically on the basis of paths
+     *
+     * @param pathConfigurations
+     *          Collection of {@link PathConfiguration} to sort
+     * @return
+     *          Sorted Collection of {@link PathConfiguration}
+     */
+    private static Collection<PathConfiguration> sortPathConfigs(
+            @Nonnull final Collection<PathConfiguration> pathConfigurations) {
+        Collection<PathConfiguration> modifiablePathConfig = pathConfigurations;
+
+        modifiablePathConfig.collect { element ->
```

Comment:
@jazeren1 @jdigger @viveksachdeva @jbornemann 
i just noticed that the original pathConfiguration's path is also changing here. i think it's because collect works on a shallow copy of modifiablePathConfig.
though i'm using collecting the returned value of the method now, still mutation of the original one before that re-assignment does not seem right somehow.

your thoughts please ?